modules/redpanda/redpanda_test.go

```diff
@@ -278,6 +293,117 @@ func TestRedpandaWithTLS(t *testing.T) {
 	require.Error(t, results.FirstErr(), kerr.UnknownTopicOrPartition)
 }
 
+func TestRedpandaListener_Simple(t *testing.T) {
+	ctx := context.Background()
+
+  	// 1. Create network 
+	RPNetwork, err := network.New(ctx, network.WithCheckDuplicate())
+	require.NoError(t, err)
+
+	// 2. Start Redpanda container
+  	// withListenerRP {
+	container, err := RunContainer(ctx,
+		testcontainers.WithImage("redpandadata/redpanda:v23.2.18"),
+		network.WithNetwork([]string{"redpanda-host"}, RPNetwork),
+		WithListener("redpanda:29092"), WithAutoCreateTopics(),
+	)
+  	// }
+	require.NoError(t, err)
+
+  	// 3. Start KCat container
+	// withListenerKcat {
+	kcat, err := testcontainers.GenericContainer(ctx, testcontainers.GenericContainerRequest{
+		ContainerRequest: testcontainers.ContainerRequest{
+			Image: "confluentinc/cp-kcat:7.4.1",
+			Networks: []string{
+				RPNetwork.Name,
+			},
+			Entrypoint: []string{
+				"sh",
+			},
+			Cmd: []string{
+				"-c",
+				"tail -f /dev/null",
+			},
+		},
+		Started: true,
+	})
+  	// }
+
+	require.NoError(t, err)
+
+	// 4. Copy message to kcat
+	err = kcat.CopyToContainer(ctx, []byte("Message produced by kcat"), "/tmp/msgs.txt", 700)
+	require.NoError(t, err)
+
+	// 5. Produce message to Redpanda
+  	// withListenerExec {
+	_, _, err = kcat.Exec(ctx, []string{"kcat", "-b", "redpanda:29092", "-t", "msgs", "-P", "-l", "/tmp/msgs.txt"})
+  	// }
+
+	require.NoError(t, err)
+
+	// 6. Consume message from Redpanda
+	_, stdout, err := kcat.Exec(ctx, []string{"kcat", "-b", "redpanda:29092", "-C", "-t", "msgs", "-c", "1"})
+	require.NoError(t, err)
+	out, err := io.ReadAll(stdout)
```

Comment:
Nit: separate concerns
```suggestion

	out, err := io.ReadAll(stdout)
```